### DISCUSSION TITLE
ESP8266 + RS485 conversion board

### DISCUSSION THREAD

**Discussion by guba91** [nathanmarlor/foxess_modbus#243] [Q&A]
Hi,
i'm thinking to use a ESP8266 or ESP32 connected to a cheap RS485 conversion board to use the foxess_modbus.

is it possible?
what code do i need to run on the ESP? esphome? this is the point that i don't understand.. should i just create a rs485 to wifi adapter using something like serial over wifi or something?

i'm really not an expert with serial protocol -.-

**Comment by calum-mcfarlane** [comment]
I'm sure you could build something to do this, but unless you want to do so as a learning exercise, worth checking the [Wiki](https://github.com/nathanmarlor/foxess_modbus/wiki) pages for devices that have already been tested to work with this integration (and others that also use Modbus to talk to Fox inverters). Some of them are available (depending on your location) for probably less than the parts you'd need to do a DIY one (unless you happen to do a lot of this sort of thing and already have a lot of parts on hand!)

**Comment by canton7** [comment]
If you're not comfortable with what you're doing, probably best to buy something cheap. Although the risk of damage to your inverter is fairly low if you roll your own a adapter, the severity is a broken inverter.
